src/OpenTelemetry/Metrics/HistogramBuckets.cs

```diff
@@ -45,6 +45,8 @@ internal HistogramBuckets(double[] explicitBounds)
 
         public Enumerator GetEnumerator() => new(this);
 
+        internal HistogramBuckets Copy() => (HistogramBuckets)this.MemberwiseClone();
```

Comment:
If you do this...

```csharp
var copy = (HistogramBuckets)this.MemberwiseClone();
var result = Object.ReferenceEquals(this.SnapshotBucketCounts, copy.SnapshotBucketCounts);
```

Is `result` == `true`?

Asking because I feel like `MemberwiseClone` will just copy the pointer, but not totally sure.

We might need to do...

```csharp
Array.Copy(this.SnapshotBucketCounts, copy.SnapshotBucketCounts, this.SnapshotBucketCounts.Length);
```

Reply:
You're right. That would also explain why MemberwiseClone had fewer allocations.
Updating the unit test to catch this.